src/format/kdbx4/parse.rs

```diff
@@ -79,16 +85,21 @@ pub(crate) fn decrypt_kdbx4(
     // derive master key from composite key, transform_seed, transform_rounds and master_seed
     let key_elements: Vec<&[u8]> = key_elements.iter().map(|v| &v[..]).collect();
     let composite_key = crypt::calculate_sha256(&key_elements)?;
-    let transformed_key = header.kdf.get_kdf().transform_key(&composite_key)?;
-    let master_key = crypt::calculate_sha256(&[header.master_seed.as_ref(), &transformed_key])?;
+    let transformed_key = outer_header
+        .kdf_settings
+        .get_kdf(&outer_header.kdf_seed)
```

Comment:
@sseemayer same comment as with `get_cipher`: We could have a `get_kdf` for parse, and a `get_fresh_kdf` that would encapsulate the IV generation.

Reply:
Good idea, added in #123